pinot-common/src/test/java/org/apache/pinot/sql/parsers/CalciteSqlCompilerTest.java

```diff
@@ -832,4 +849,17 @@ public void testSqlDistinctQueryCompilation() {
     Assert.assertEquals(aggregationInfo.getAggregationParams().get(FunctionCallAstNode.COLUMN_KEY_IN_AGGREGATION_INFO),
         "add(div(col1,col2),mul(col3,col4)):sub(col3,col4):col5:col6");
   }
+
+  @Test
+  public void testQueryValidationFailure() {
+    String sql;
+    try {
+      sql = "select group_city,group_country, sum(rsvp_count), count(*) from meetupRsvp group by group_country, sum(rsvp_count), count(*) limit 50";
```

Comment:
I agree that we should be checking for identifiers. However, I don't think a query like this:
SELECT State, Count(*) from Customers GROUP BY State, Count(*) should be supported.
In other words, we should not have the aggregations as one of the GROUP BY columns.

Reply:
Try the following queries here https://www.w3schools.com/sql/trysql.asp?filename=trysql_select_no_distinct

(1) SELECT City, SUM(PostalCode) FROM Customers GROUP BY City, SUM(PostalCode)
(2) SELECT City, SUM(PostalCode) FROM Customers GROUP BY City

(2) is valid whereas (1) fails

Reply:
So we should also add tests for this case too

Reply:
The error seen on executing the above examples as per the w3schoold website is:

Error 1: could not prepare statement (1 aggregate functions are not allowed in the GROUP BY clause).

Reply:
GROUP BY clause shouldn't contain aggregate expression like sum(PostalCode), however, you can still do dateConverter(secondsSinceEpoch) etc, which is non-aggregate Transform function. Or expressions like ADD(a,b). 

My validate assumes that group by clause is always non-aggregate expression, so that we could extract all the identifier from them.

Reply:
Also added a validation to prevent aggregate expression in group by clause.